apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx

```diff
@@ -46,185 +39,209 @@ interface CustomEventDialogProps {
+    const [start, setStart] = useState(defaultCustomEventValues.start);
+    const [end, setEnd] = useState(defaultCustomEventValues.end);
+    const [title, setTitle] = useState(defaultCustomEventValues.title);
+    const [days, setDays] = useState(defaultCustomEventValues.days);
 
-    handleOpen = () => {
-        this.setState({ open: true, scheduleIndices: [AppStore.schedule.getCurrentScheduleIndex()] });
+    useEffect(() => {
+        const handleSkeletonModeChange = () => {
+            setSkeletonMode(AppStore.getSkeletonMode());
+        };
+
+        AppStore.on('skeletonModeChange', handleSkeletonModeChange);
+
+        return () => {
+            AppStore.off('skeletonModeChange', handleSkeletonModeChange);
+        };
+    }, []);
+
+    const handleOpen = useCallback(() => {
+        setOpen(true);
+        setScheduleIndices([AppStore.schedule.getCurrentScheduleIndex()]);
 
         logAnalytics({
             category: analyticsEnum.calendar.title,
             action: analyticsEnum.calendar.actions.CLICK_CUSTOM_EVENT,
         });
-    };
+    }, [setOpen]);
 
-    handleClose = (cancel: boolean) => {
-        if (!cancel) {
-            logAnalytics({
-                category: analyticsEnum.calendar.title,
-                action: analyticsEnum.calendar.actions.ADD_CUSTOM_EVENT,
-            });
-            if (this.props.onDialogClose) this.props.onDialogClose();
-            this.handleAddToCalendar();
-        }
-
-        this.setState({
-            ...this.state,
-            open: false,
-        });
-    };
-
-    handleEventNameChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
-        this.setState({ title: event.target.value });
-    };
+    const handleClose = useCallback(() => {
+        setOpen(false);
+    }, [setOpen]);
 
-    handleEndTimeChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
-        this.setState({ end: event.target.value });
-    };
+    const handleSubmit = () => {
+        handleClose();
+        handleAddToCalendar();
 
-    handleStartTimeChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
-        this.setState({ start: event.target.value });
+        logAnalytics({
+            category: analyticsEnum.calendar.title,
+            action: analyticsEnum.calendar.actions.ADD_CUSTOM_EVENT,
+        });
     };
 
-    handleDayChange = (days: boolean[]) => {
-        this.setState({ days: days });
+    const resetForm = () => {
+        setStart(defaultCustomEventValues.start);
+        setEnd(defaultCustomEventValues.end);
+        setTitle(defaultCustomEventValues.title);
+        setDays(defaultCustomEventValues.days);
     };
 
-    handleAddToCalendar = () => {
-        if (!this.state.days.some((day) => day) || this.state.scheduleIndices.length === 0) return;
+    const handleAddToCalendar = () => {
+        if (!days.some((day) => day) || scheduleIndices.length === 0) return;
```

Comment:
```suggestion
        if (!days.includes(true) || scheduleIndices.length === 0) return;
```
Would this be make more sense?